src/imageio/format/jpeg.c

```diff
@@ -396,6 +394,10 @@ int write_image(dt_imageio_module_data_t *jpg_tmp, const char *filename, const v
   jpeg_finish_compress(&(jpg->cinfo));
   jpeg_destroy_compress(&(jpg->cinfo));
   fclose(f);
+
+  if(exif && exif_len > 0 && exif_len < 65534)
```

Comment:
I'm having a look at this issue (since I feel responsible), but it seems that there's something wrong at the level of the data itself: the loop over tags ( src/common/exif.cc:1081) do not even do a single iteration.

There's an exception apprently, which I don't have on my own Pentax files:
[exiv2] /home/mazhe/tmp/foo.jpg: This does not look like a TIFF image